### ISSUE TITLE
FileNotFoundException is thrown serving resources due to JarUrlConnection.getLastModified() returning zero

### ISSUE THREAD

**Issue by adora-tech** [spring-projects/spring-boot#38204]
Hello,

First, this problem does not appear in the case of Spring Boot 3.1.5
What I do here is to serve static content from the `src\main\resources\static` path.
Static content is generated with angular-cli v17, which is a simple project.

I created two simple examples, one with Spring Boot 3.1.5 in which the problem does not appear and works correctly,
and one with Spring Boot 3.2.0 RC2 in which this problem can be observed.

This problem only occurs after the project is assembled with `.\gradlew clean assemble` and is run with `java -jar .\app-sample-sb320-rc2-issue-0.0.1-SNAPSHOT.jar`

In the project app-sample-sb320-rc2-issue.zip, you will find in the root, the `issue-details` directory, which in turn contains files:

browser-access-issue.png
![browser-access-issue](https://github.com/spring-projects/spring-boot/assets/58594685/cae0c019-0f3f-4cd2-ac2d-a03448b7540f)

[log-acces-issue.txt](https://github.com/spring-projects/spring-boot/files/13251686/log-acces-issue.txt)

Attached you will find the two archived projects:

[app-sample-sb315-working.zip](https://github.com/spring-projects/spring-boot/files/13251704/app-sample-sb315-working.zip)

[app-sample-sb320-rc2-issue.zip](https://github.com/spring-projects/spring-boot/files/13251705/app-sample-sb320-rc2-issue.zip)

Thank you,
Iulius Ciorica

**Comment by wilkinsona** [comment]
Thanks for the report and sample app. This appears to be a regression caused by the new nested jar support. The problem does not occur if you switch back to the old support by adding the following to `build.gradle`:

```groovy
tasks.named('bootJar') {
    loaderImplementation = org.springframework.boot.loader.tools.LoaderImplementation.CLASSIC
}
```

**Comment by philwebb** [comment]
@adora-tech Thanks so much for finding and raising this one before we released 3.2.0. The sample was super helpful and I think I've identified and fixed the underlying problem.
